src/main/java/pl/bootcamp/ex21_1/controller/ProductController.java

```diff
@@ -0,0 +1,53 @@
+package pl.bootcamp.ex21_1.controller;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import pl.bootcamp.ex21_1.model.Product;
+import pl.bootcamp.ex21_1.model.ProductRepository;
+
+import java.util.List;
+
+@Controller
+public class ProductController {
+
+    private final ProductRepository productRepository;
+
+    public ProductController(ProductRepository productRepository) {
+        this.productRepository = productRepository;
+    }
+
+    @GetMapping("/")
+    public String goHome() {
+        return "redirect:/front.html";
+    }
+
+    @GetMapping("/list")
+    public String searchProductsForGivenCategory(Model model, @RequestParam(required = false, name = "kategoria") String polishCategory) {
+        List<Product> products;
+        if (polishCategory != null) {
+            products = productRepository.findByCategory(polishCategory);
+        } else {
+            products = productRepository.getAll();
+        }
+        model.addAttribute("products", products);
+        var totalPrice = productRepository.getTotalPrice(products);
```

Comment:
Do repozytorium raczej sie tak nie przekazuje do wczesniej pobranej listy, mozesz sume bezposrednio wyliczyc w kontrolerze (nie jest to też do końca poprawne, ale na ten moment nam wystarczy)